python/modules/IcePy/Proxy.cpp

```diff
@@ -1620,18 +1621,93 @@ checkedCastImpl(ProxyObject* p, const string& id, PyObject* facet, PyObject* ctx
     return Py_None;
 }
 
+static PyObject*
```

Comment:
I don't understand why we implement Python's checkedCast (and now checkCastAsync) in C++.

Why not a simple wrapper (in pure Python) over the Python's ice_isAAsync?

Reply:
Yes, it might be simpler to rework this and define both in Python.

Reply:
Reworked to move checkedCast implementation to ObjectPrx.py